source/fuzz/fuzzer_util.h

```diff
@@ -287,6 +287,16 @@ std::vector<opt::Instruction*> GetCallers(opt::IRContext* ir_context,
 opt::Function* GetFunctionFromParameterId(opt::IRContext* ir_context,
                                           uint32_t param_id);
 
+// Creates a OpTypeFunction instruction with result id
+// |new_function_type_result_id| and operands |new_function_type_operands|.
+// Assigns the created function type as the type of the function with result id
+// |function_id|. Reuses the old type of the function if possible. Returns the
+// result id of the type, the function uses.
+uint32_t MaybeReuseFunctionType(
+    opt::IRContext* ir_context, uint32_t function_id,
+    uint32_t new_function_type_result_id,
+    const std::vector<uint32_t>& new_function_type_operands);
```

Comment:
I think the name and comments could be clearer.

How about UpdateFunctionType as a name?

And as a comment, something along the lines of:

  Changes the type of function |function_id| so that its return type is |return_type_id| and its operand types are |operand_type_ids|.  If a suitable function type already exists in the module, it is used, otherwise |new_function_type_result_id| - which must be fresh - is used as the result id of a suitable new function type instruction.

Reply:
Fixed in e5fda6e. Although, I don't think `new_function_type_result_id` should be fresh. The reason I can think of is the same as for the other `fuzzerutil::AddConstant*` and `fuzzerutil::AddType*` functions - we might want to reuse the id (e.g. in one of the transformations I worked on, we reused the id for the global variable). @afd, what do you think about this?

Reply:
... should necessarily be fresh...